temba/triggers/models.py

```diff
@@ -271,7 +275,11 @@ def catch_triggers(cls, entity, trigger_type, channel, referrer_id=None, extra=N
             triggers = triggers.filter(models.Q(channel=channel) | models.Q(channel=None))
 
         if referrer_id is not None:
-            triggers = triggers.filter(referrer_id=referrer_id)
+            triggers = triggers.filter(models.Q(referrer_id=referrer_id) | models.Q(referrer_id=''))
```

Comment:
agreed, I also needed to update here: https://github.com/rapidpro/rapidpro/blob/master/temba/triggers/views.py#L317